internal/config/cardpick.go

```diff
@@ -0,0 +1,104 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"log"
+	"strconv"
+	"strings"
+
+	"github.com/vinegarhq/vinegar/sysinfo"
+)
+
+// Check if the system actually has PRIME offload and there's no ambiguity with the GPUs.
+func primeIsAllowed(isVulkan bool) (bool, error) {
+	//There's no ambiguity when there's only one card.
+	if len(sysinfo.Cards) <= 1 {
+		log.Printf("Number of cards is equal or below 1. Skipping prime logic.")
+		return false, nil
+	}
+	//Check if the main card has an embedded display.
+	if !sysinfo.Cards[0].Embedded {
+		log.Printf("card0 is not embedded. This machine is not a laptop. Skipping prime logic.")
+		return false, nil
+	}
+	//Don't mess with devices that have more than two cards.
+	if len(sysinfo.Cards) > 2 {
+		//OpenGL cannot choose the right card properly. Prompt user to define it themselves
+		if !isVulkan {
+			return false, fmt.Errorf("opengl cannot select the right gpu. gpus detected: %d", len(sysinfo.Cards))
+		}
+		log.Printf("System has %d cards. Skipping prime logic and leaving card selection up to Vulkan.", len(sysinfo.Cards))
+		return false, nil
+	}
+	return true, nil
+}
+
+func pickCard(opt string, env Environment, isVulkan bool) error {
+	if opt == "" { //Default value for opt
+		opt = "none"
+	}
```

Comment:
This is not going to do anything, just check if it is empty and immediately return, as on line 57 you already immediately return with no changes.

Reply:
So, instead of this if statement, include an empty string as a case?

Reply:
I'm so stupid. Will apply change.